db/db_utils.go

```diff
@@ -90,6 +90,32 @@ func InsertManyUnordered(c string, items ...interface{}) error {
 	return errors.WithStack(err)
 }
 
+// CreateCollections ensures that all the given collections are created,
+// returning an error immediately if creating any one of them fails.
+func CreateCollections(collections ...string) error {
+	session, db, err := GetGlobalSessionFactory().GetSession()
+	if err != nil {
+		return err
+	}
+	defer session.Close()
+
+	const namespaceExistsErrCode = 48
+	for _, collection := range collections {
+		_, err := db.CreateCollection(collection)
+		if err == nil {
+			continue
+		}
+		// If the collection already exists, this does not count as an error.
+		if mongoErr, ok := errors.Cause(err).(mongo.CommandError); ok && mongoErr.HasErrorCode(namespaceExistsErrCode) {
```

Comment:
What are your thoughts on matching against the error code vs. [listing the existing collection names](https://pkg.go.dev/go.mongodb.org/mongo-driver/mongo#Database.ListCollectionNames) and only creating the ones that don't already exist?

Reply:
I prefer not to do that since this is a stable error code that proves in an atomic way that at the moment of the creation attempt, it already existed. Also, list + create adds an extra DB operation and would require more Anser changes.

Reply:
sgtm